core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -132,7 +132,7 @@ class OptimisticTransaction
   extends OptimisticTransactionImpl
   with DeltaLogging {
 
-  assertNoDeletionVectors(snapshot.metadata, spark)
+  DeletionVectorUtils.assertDeletionVectorsNotReadable(spark, snapshot.metadata, snapshot.protocol)
```

Comment:
Here we can simply check if the protocol contains the DV table feature. The snapshot constructor did make sure the protocol and metadata are in sync.